### PR TITLE
Feature: Playlist pagination

### PR DESCRIPTION
**Pull request by bcmrbvm**
This PR allows the user to load more tracks if the playlist is longer than 20 tracks

### CODE REVIEW

File: app/js/playlist/controllers/PlaylistCtrl.js [hunk 1]
```diff
@@ -100,7 +132,7 @@ angular.module("FM.playlist.PlaylistCtrl", [
          * @method onEnd
          */
         $scope.onEnd = function onEnd(event, data) {
-            if ($scope.playlist[0].track.uri === data.uri) { // jshint ignore:line
+            if (($scope.playlist[0].track.uri === data.uri) && ($scope.page.pages === $scope.page.total)) { // jshint ignore:line
                 $scope.meta.total--;
                 $scope.meta.play_time = $scope.meta.play_time - $scope.playlist[0].track.duration; // jshint ignore:line
```
Comment (jamesjwarren) [inline app/js/playlist/controllers/PlaylistCtrl.js]: As queue meta is not paginated, this should always be updated onEnd, only splicing the track from the list needs to be wrapped in this pagination condition.

Comment (bcmrbvm) [inline app/js/playlist/controllers/PlaylistCtrl.js]: my reason for wrapping all of this in the pagination condition was that if there are more pages of tracks then the user should always have at least the first 20 tracks. Incase they never load more tracks and it go down to one track but there are still more tracks which where never loaded from the second page.

If there is only one page of tracks we can handle all the logic in the FE.

Comment (jamesjwarren) [inline app/js/playlist/controllers/PlaylistCtrl.js]: Ahh I see, that makes sense now.

**Comment by bcmrbvm** [comment]
@jamesjwarren is it okay to merge then?

**Comment by jamesjwarren** [comment]
LGTM :+1:

**Pull request merged**